src/main/java/echoserver/EchoClient.java

```diff
@@ -3,41 +3,29 @@
 import java.net.*;
 
 public class EchoClient {
-    private static int port = 8080;
-    private static String host;
-    private static Socket clientSocket;
-
-    public static void main(String[] args) {
-        host = args[0];
-        port = Integer.parseInt(args[1]);
-
-        if (args.length != 2) {
-            System.err.println("Usage: java EchoClient <host name> <port number>");
-            System.exit(1);
-        }
-
-        try {
-            ClientSocketWrapper clientSocketWrapper = new ClientSocketWrapper();
-            startEchoClient(clientSocketWrapper);
-        } catch (IOException e) {
-            System.out.println("Cannot create client socket");
-        }
+    int port;
+    InetAddress host;
+    ClientWrapper socketWrapper;
+
+    public EchoClient(int port, InetAddress host, ClientWrapper socketWrapper) {
+        this.port = port;
+        this.host = host;
+        this.socketWrapper = socketWrapper;
     }
 
-    public static void startEchoClient(ClientSocketWrapper clientSocketWrapper) throws IOException {
+    public void startEchoClient() {
+        String clientData;
         try {
-            clientSocket = clientSocketWrapper.startClientSocket(port, host);
-            System.out.println("Connection successful using host " + host + " on " + port);
-            String clientData;
-
-            while ((clientData = clientSocketWrapper.receiveData()) != null) {
-                clientSocketWrapper.sendData(clientData);
-
-                if (clientSocketWrapper.quit(clientData)) {
-                    clientSocketWrapper.close();
+            socketWrapper.startClientSocket(host, port);
+            while (true) {
```

Comment:
I had it set up without the outer while loop and it would close out the connection as soon as it happened because there was no input coming from the client. This only happens on the terminal, not within IntelliJ.